beacon-chain/slasher/detect_attestations.go

```diff
@@ -71,6 +81,69 @@ func (s *Service) detectSlashableAttestations(
 	return s.serviceCfg.Database.SaveLatestEpochAttestedForValidators(ctx, validatorIndices, args.currentEpoch)
 }
 
+// Check for attester slashing double votes by looking at every single validator index
+// in each attestation's attesting indices and checking if there already exist records for such
+// attestation's target epoch. If so, we append a double vote slashing object to a list of slashings
+// we return to the caller.
+func (s *Service) checkDoubleVotes(
+	ctx context.Context, attestations []*slashertypes.CompactAttestation,
+) ([]*slashertypes.Slashing, error) {
+	// We check if there are any slashable double votes in the input list
+	// of attestations with respect to each other.
+	slashings := make([]*slashertypes.Slashing, 0)
+	existingAtts := make(map[string][32]byte)
+	for _, att := range attestations {
+		for _, valIdx := range att.AttestingIndices {
+			key := fmt.Sprintf("%d:%d", att.Target, valIdx)
```

Comment:
`Sprintf` is the slowest among different ways concatenate. Given the large number of attesting indices per slot, we should be frugal here.

https://dev.to/pmalhaire/concatenate-strings-in-golang-a-quick-benchmark-4ahh